File: src/main/java/net/omori_sunny/create_waterparked/game/contraption/AnchorPeerCurveDataAccess.java
package net.omori_sunny.create_waterparked.game.contraption;

import com.simibubi.create.content.trains.track.BezierConnection;
import net.minecraft.core.BlockPos;

import java.util.Map;

// read and write access to the inherited peer curve maps
public interface AnchorPeerCurveDataAccess {
    Map<BlockPos, BezierConnection> waterparked$anchorPeerCurves();

    Map<BlockPos, Integer> waterparked$railRgb();

    Map<BlockPos, Integer> waterparked$beamRgb();
}

File: src/main/kotlin/net/omori_sunny/create_waterparked/game/contraption/AnchorPeerCurveDataAccess.kt
package net.omori_sunny.create_waterparked.game.contraption

import com.simibubi.create.content.trains.track.BezierConnection
import net.minecraft.core.BlockPos

// read and write access to the inherited peer curve maps
interface AnchorPeerCurveDataAccess {
    fun `waterparked$anchorPeerCurves`(): MutableMap<BlockPos, BezierConnection>
    fun `waterparked$railRgb`(): MutableMap<BlockPos, Int>
    fun `waterparked$beamRgb`(): MutableMap<BlockPos, Int>
}
